test/boulder-tools/Dockerfile

```diff
@@ -11,13 +11,12 @@ COPY requirements.txt /tmp/requirements.txt
 COPY boulder.rsyslog.conf /etc/rsyslog.d/
 COPY build.sh /tmp/build.sh
 RUN /tmp/build.sh
+
 # Note: This arg and env variable should only be set _after_ build.sh installs
 # necessary apt packages. Otherwise, we will reinstall those apt packages for
 # each Go version, rather than reusing a cached layer.
 ARG GO_VERSION
 ENV GO_VERSION=$GO_VERSION
 COPY install-go.sh /tmp/install-go.sh
-RUN /tmp/install-go.sh
-RUN sed -i '/imklog/s/^/#/' /etc/rsyslog.conf
-RUN sed -i '/$ActionFileDefaultTemplate/s/^/#/' /etc/rsyslog.conf
-RUN sed -i '/$RepeatedMsgReduction on/s/^/#/' /etc/rsyslog.conf
+RUN /tmp/install-go.sh && \
```

Comment:
Why combine the rsyslog editing steps with the install-go step? Though I do like combining the different sed invocations into a single invocation.

Reply:
`RUN`, `COPY`, and `ADD` create new layers which use some amount of disk space. Having less layers is more efficient disk space-wise. There's further optimizations that can be done with a multi-stage build, but that's outside the scope of this PR. Hell, even this change is now outside the scope of this PR because I changed strategies mid-way through.

Reply:
Layers prior to this change
![Screenshot from 2023-07-10 16-40-26](https://github.com/letsencrypt/boulder/assets/2382565/82a11d94-6d70-4299-aba6-388a443f1598)

Layers after this change
![Screenshot from 2023-07-10 16-49-52](https://github.com/letsencrypt/boulder/assets/2382565/2da96f5b-ddb0-49a1-96c9-6d18c988d57d)

Reply:
Good point, and that's why for instance we put a bunch of stuff inside `build.sh` rather than in RUN commands directly. Perhaps the `sed` commands should go inside `build.sh`? That makes more sense to me logically than clustering them with `install-go.sh`. It would probably be interesting to look at the git history and see if there's a specific reason they're in the Dockerfile rather than build.sh.